flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkDistribution.java

```diff
@@ -301,7 +309,105 @@ public void submitSQLJob(SQLJobSubmission job, Duration timeout) throws Exceptio
                             });
             new Thread(future).start();
             future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+        } else if (job.getClientMode() == SQLJobSubmission.ClientMode.REST) {
+            FutureTaskWithException<Void> future =
+                    new FutureTaskWithException<>(
+                            () -> {
+                                Map<String, String> configMap =
+                                        GlobalConfiguration.loadConfiguration(
+                                                        conf.toAbsolutePath().toString())
+                                                .toMap();
+                                String host =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.address",
+                                                InetAddress.getByName("localhost")
+                                                        .getHostAddress());
+                                String port =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.port", "8083");
+                                // Open a session
+                                String sessionHandle = openSession(host, port);
+                                List<String> sqlLines = job.getSqlLines();
+                                for (String sql : sqlLines) {
+                                    // Execute statement
+                                    String operationHandle =
+                                            executeStatement(sessionHandle, sql, host, port);
+                                    int count = 5;
+                                    while (count > 0) {
+                                        TimeUnit.SECONDS.sleep(2);
+                                        // Fetch results
+                                        String results =
+                                                fetchResults(
+                                                        sessionHandle, operationHandle, host, port);
+                                        if (results.contains("PAYLOAD")) {
+                                            break;
+                                        }
+                                        --count;
+                                    }
+                                }
+                            });
+            new Thread(future).start();
+            future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
Can we add such method to reuse the logic in the HIVE_JDBC and REST?


```
 private void submitSQL(RunnableWithException command, Duration timeout) {
        FutureTaskWithException<Void> future = new FutureTaskWithException<>(command);
        new Thread(future).start();
        future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
    }
```

Reply:
Done.